packages/featureflags/featureToggleManager.js

```diff
@@ -7,89 +7,101 @@ const { getRedisClient } = require('@open-condo/keystone/redis')
-            logger.error({ msg: 'fetchFeatures error', error: e })
-
-            return {}
+        } else if (this._static) {
+            return JSON.parse(JSON.stringify(this._static))
         }
-    }
 
-    async #getContext (keystoneContext) {
-        return get(keystoneContext, 'req', null)
-            ? keystoneContext
-            : { req: { features: await this.fetchFeatures() } }
+        throw new Error('FeatureToggleManager config error!')
     }
 
-    async isFeatureEnabled (keystoneContext, featureName, featuresContext) {
-        const context = await this.#getContext(keystoneContext)
-        const request = context.req
-        const headersFeatureFlags = get(request, ['headers', 'feature-flags'])
-
-        // Here it will stop under tests
-        if (conf.NODE_ENV === 'test') return headersFeatureFlags === 'true'
+    async #getFeaturesFromKeystoneContext (keystoneContext) {
+        const req = get(keystoneContext, 'req')
+        let features = get(req, 'features')
 
-        const growthbook = new GrowthBook()
+        // Note: fetch features if needed! And save it in `req` if in request context
+        if (!features) {
+            features = await this.fetchFeatures()
+            if (req) req.features = features
+        }
 
-        growthbook.setFeatures(request.features)
+        return features
+    }
 
+    async #getGrowthBookInstance (keystoneContext, featuresContext) {
+        const features = await this.#getFeaturesFromKeystoneContext(keystoneContext)
+        const growthbook = new GrowthBook()
+        growthbook.setFeatures(features)
         if (featuresContext) growthbook.setAttributes(featuresContext)
+        return growthbook
+    }
+
+    async isFeatureEnabled (keystoneContext, featureName, featuresContext) {
+        // Note: if you want to override the flag value by tests you cen set the `feature-flags` header! (TESTS ONLY)
+        if (conf.NODE_ENV === 'test') {
+            const headersFeatureFlags = get(keystoneContext, ['req', 'headers', 'feature-flags'])
+            return headersFeatureFlags === 'true'
+        }
 
+        const growthbook = await this.#getGrowthBookInstance(keystoneContext, featuresContext)
         return growthbook.isOn(featureName)
     }
 
     async getFeatureValue (keystoneContext, featureName, defaultValue, featuresContext) {
-        const context = await this.#getContext(keystoneContext)
-        const request = context.req
-
-        // Here it will stop under tests
-        if (conf.NODE_ENV === 'test') return defaultValue
-
-        const growthbook = new GrowthBook()
-
-        growthbook.setFeatures(request.features)
-
-        if (featuresContext) growthbook.setAttributes(featuresContext)
+        // Note: if you want to override the flag value by tests you cen set the `feature-value` header! (TESTS ONLY)
+        if (conf.NODE_ENV === 'test') {
+            const headersFeatureValue = get(keystoneContext, ['req', 'headers', 'feature-value'])
```

Comment:
Following naming feels more explicit and consistent:

```suggestion
            const headersFeatureValue = get(keystoneContext, ['req', 'headers', 'feature-flags'])
```

Reply:
no feture flags and feature values is different